src/shared/components/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { Button as BsButton } from "react-bootstrap";
+import type { ButtonProps as BsButtonProps } from "react-bootstrap";
+import { Link } from "react-router-dom";
+import type { LinkProps } from "react-router-dom";
+
+import "./styles/Button.scss";
+
+export interface ButtonProps {
+  variant:
+    | "dark"
+    | "light"
+    | "pink-dark"
+    | "pink-light"
+    | "grey-dark"
+    | "grey-light";
+}
+
+const Button = React.forwardRef<
+  HTMLButtonElement,
+  ButtonProps & Omit<BsButtonProps, "variant"> & JSX.IntrinsicElements["button"]
+>(({ variant, className, children, ...props }, ref) => {
+  return (
+    <BsButton
+      ref={ref}
+      variant="primary"
+      {...(props as BsButtonProps)}
+      className={`Button Button-${variant} ${className || ""}`}
+    >
+      {children}
+    </BsButton>
+  );
+});
+
+export const LinkButton = React.forwardRef<
+  HTMLAnchorElement,
+  ButtonProps & LinkProps
+>(({ variant, className, children, ...props }, ref) => {
+  return (
+    <Link
```

Comment:
Would it be possible to use the `as` prop on the react-bootstrap button for this use case?

Reply:
I've tried, but it breaks the typings on refs.